### PR TITLE
Elasticsearch: base tests

### PR DESCRIPTION
**Pull request by kvignesh1420**
This PR addresses the feature request: https://github.com/tensorflow/io/issues/1093 by adding the base test cases.

### CODE REVIEW

File: tests/test_elasticsearch_eager.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+    BATCH_SIZE = 2
+    dataset = tfio.experimental.elasticsearch.ElasticsearchIODataset(
+        nodes=[NODE], index=INDEX, doc_type=DOC_TYPE
+    ).batch(BATCH_SIZE)
+
+    assert issubclass(type(dataset), tf.data.Dataset)
+
+    for item in dataset:
+        for attr in ["name", "gender", "age", "fare", "survived"]:
```
Comment (terrytangyuan) [inline tests/test_elasticsearch_eager.py:116]: Move this list to a constant since it appears in multiple places

**Comment by kvignesh1420** [comment]
@terrytangyuan I switched to using sockets for checking the presence of the es service as docker was posing trouble. Also, moved the `ATTRS` to a common var

File: tests/test_elasticsearch_eager.py (hunk 1 above)
Comment (kvignesh1420) [reply tests/test_elasticsearch_eager.py]: addressed in the commit

**Review by terrytangyuan — APPROVED** [review]
LGTM. Others PTAL

**Review by yongtang — APPROVED** [review]
LGTM

**Pull request merged**